hack/make/deploy/kubernetes.mk

```diff
@@ -1,24 +1,32 @@
-## Deploy the operator in the Kubernetes cluster configured in ~/.kube/config
-deploy/kubernetes: manifests/crd/helm
-	kubectl get namespace dynatrace || kubectl create namespace dynatrace
-	helm template dynatrace-operator config/helm/chart/default \
+ENABLE_CSI ?= true
+
+## Deploy the operator in a cluster configured in ~/.kube/config where platform and version are autodetected
+deploy: manifests/crd/helm
+	helm upgrade dynatrace-operator config/helm/chart/default \
 			--namespace dynatrace \
+			--create-namespace \
+			--install \
+			--atomic \
 			--set installCRD=true \
-			--set platform="kubernetes" \
-			--set csidriver.enabled=true \
+			--set csidriver.enabled=$(ENABLE_CSI) \
 			--set manifests=true \
-			--set image="$(IMAGE_URI)" | kubectl apply -f -
+			--set image="$(IMAGE_URI)"
```

Comment:
why did we keep the old `deploy/kubernetes` (and others) entrypoints ?
if we are going to keep them, shouldn't they act the same? (use helm upgrade so they are managed by helm)
also the openshift entrypoints were not updated, why? (they don't use `$(ENABLE_CSI)`)

Reply:
I kept the entries on purpose to give us the possibility to still deploy/undeploy the operator without helm from the makefile.

Will rename `deploy` to `deploy/helm` and move it in it's own file. 

Concerning the openshift entries, I will adapt them too